x-pack/plugin/esql/qa/server/single-node/src/yamlRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/EsqlClientYamlIT.java

```diff
@@ -22,6 +24,17 @@ public EsqlClientYamlIT(final ClientYamlTestCandidate testCandidate) {
 
     @ParametersFactory
     public static Iterable<Object[]> parameters() throws Exception {
-        return createParameters();
+        return updateEsqlQueryDoSections(createParameters(), EsqlClientYamlIT::modifyEsqlQueryExecutableSection);
     }
+
+    private static ExecutableSection modifyEsqlQueryExecutableSection(DoSection doSection) {
+        var apiCallSection = doSection.getApiCallSection();
+        if (apiCallSection.getApi().equals("esql.query")) {
+            if (apiCallSection.getParams().containsKey("allow_partial_results") == false) {
+                apiCallSection.addParam("allow_partial_results", "false"); // we want any error to fail the test
```

Comment:
Should we do this in the other esql yaml subclasses?

I'm a bit worried that this makes our tests unlike production. If we're hacking the yaml files, maybe we should add an assertion that there aren't any partial failures instead of swapping this key.

@dnhatn, you wanted some time for this setting to test, what do you think?

Reply:
Thanks for the ping. I will take a look at the new PR.